fetcher/bundle.go

```diff
@@ -0,0 +1,206 @@
+package fetcher
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/dcos/dcos-diagnostics/api/rest"
+	"github.com/dcos/dcos-diagnostics/dcos"
+	"github.com/dcos/dcos-diagnostics/util"
+	"github.com/sirupsen/logrus"
+)
+
+// BundleRequest represents a request for a node's diagnostic bundle
+type BundleRequest struct {
+	BundleID string
+	Node     *dcos.Node
+	Client   *http.Client
+	port     int
+	forceTLS bool
+	Data     []byte
```

Comment:
This are response data so it means `BundleRequests` is also `BundleResponse`. Weird, I'd like to have separated structure for that.